dynamo/tools/clustering.py

```diff
@@ -174,19 +174,20 @@ def hdbscan(
 
 
 def leiden(
-    adata,
-    use_weight=True,
-    weight=None,
-    initial_membership=None,
-    adj_matrix=None,
-    adj_matrix_key=None,
-    result_key=None,
-    layer=None,
-    obsm_key=None,
-    selected_cluster_subset: list = None,
-    selected_cell_subset=None,
-    directed=False,
-    copy=False,
+    adata: AnnData,
+    resolution: float = 1.0,
+    use_weight: bool = True,
+    weight: list = None,
+    initial_membership: list = None,
+    adj_matrix: Union[list, np.array, csr_matrix, None] = None,
```

Comment:
do you support the use of list of list np.array and csr_matrix here? 

again, use Optional if None can be used

Reply:
need the fix all the following. 

it should be Optional[str] = None, etc.